bot/app/__init__.py

```diff
@@ -25,60 +26,135 @@
 from app.database import db_objects
 
 # load env variables from .env
-load_dotenv()
 
-app = Flask(__name__)
+logging.basicConfig(filename=os.path.abspath('../log.txt'), level=logging.DEBUG)
 
+load_dotenv()
 API_TOKEN = os.getenv('API_TOKEN')
+
+app = Flask(__name__)
 bot = telebot.TeleBot(API_TOKEN)
 
-img_dir = os.path.abspath('../bot/img')
+img_dir = os.path.abspath('../bot/app/img')
 
 @bot.message_handler(commands=['start', 'help'])
 def start(message):
 
+	"""Start and help message handler function
+
+	Upon sending /start or /help all available commands are returned. 
+	In the background the user's id is checked against the db and added,
+	if there is no entry for the user yet.
+
+	Arguments:
+		message: telebot's message object
+	"""
+
 	u_id = message.from_user.id
 	name = message.from_user.first_name
 	is_bot = message.from_user.is_bot
 
 	user = db_objects.User(u_id, name, is_bot)
 
 	if not user.user_exists(u_id):
-		user.add_user(u_id, name, is_bot)
-
-	bot.send_message(chat_id=u_id, text=render_template('start.html', name=name), \
-		parse_mode='html')
+		user.add_user()
 
-@bot.message_handler(commands=['test'])
-def test(message):
-	with open(os.path.join(template_dir, 'test.html'), 'r') as f:
-		reply = f.read()
-	bot.send_message(chat_id=message.chat.id, text=reply, parse_mode='html')
+	with app.app_context():
+		bot.send_message(chat_id=u_id, text=render_template('start.html', name=name), \
+			parse_mode='html')
 
-# returns flyer and info on the next event when user sends /next
 @bot.message_handler(commands=['next'])
 def next_event(message):
-	with open(os.path.join(template_dir,  'next_event.html'), 'r') as f:
-		photo_caption = f.read()
-	with open(os.path.join(img_dir, 'next_event.jpg'), 'rb') as p:
-		picture=p.read()
-	bot.send_photo(chat_id=message.chat.id, photo=picture, caption=photo_caption, parse_mode='html')
-
-@bot.message_handler(commands=['message_to_all'])
-def push_message_to_all(message):
-	with open(os.path.join(template_dir, 'test.html'), 'r') as f:
-		reply = f.read()
-	uids = db.all_users()
-	for uid in uids:
-		print(uid[0])
-		if (uid != 123456789) and (uid != 987654321):
-			bot.send_message(chat_id=uid[0], text=reply, parse_mode='html')
+
+	"""/next message handler function
+
+	Upon sending the /next command the event's graphic and all available
+	info from the db are returned.
+	Artists are represented with hyperlinks. By pressing one of those,
+	the artist message handler is triggered
+
+	Arguments:
+		message: telebot's message object
+	"""
+
+	with app.app_context():
+		u_id = message.from_user.id
+	
+		try:
+			next_event = db_objects.get_next_event()
+		except:
+			return render_template('none.html')
```

Comment:
Are you really sure to return some type of `404` message here? If you have no event planned, then it should probably say something along the lines: `No event planned yet, stay tuned for more in the future!`. Otherwise your users might think there is a problem instead of just no event being on the horizon yet.

Reply:
substituted with a more generic message, but this is not our focus right now, we can care about this, once the bot is running solid.